src/server/passive_tran_server.cpp

```diff
@@ -152,6 +166,22 @@ void passive_tran_server::start_log_replicator (const log_lsa &start_lsa)
   m_replicator.reset (new cublog::replicator (start_lsa, OLD_PAGE_IF_IN_BUFFER_OR_IN_TRANSIT, 0));
 }
 
+void passive_tran_server::send_and_receive_stop_log_prior_dispatch ()
+{
+  // empty message request
+  push_request (tran_to_page_request::SEND_STOP_LOG_PRIOR_DISPATCH, std::string ());
+
+  // wait for a confirmation to avoid race conditions where log prior is still being transferred
+  // while the passive transaction server thinks otherwise
+  {
+    std::unique_lock<std::mutex> ulock { m_confirm_log_prior_dispatch_stopped_mtx };
```

Comment:
I implemented this the "old" way.
Maybe it should be re-implemented using new send-receive implementation before actual merge.

Reply:
Yep, best to use send_receive; I can confirm it works :)